src/App.jsx

```diff
@@ -52,22 +57,29 @@ function App() {
 
   const app = useFirebaseApp();
 
+  if (import.meta.env.DEV) {
+    console.log('connecting to functions emulator');
+    connectFunctionsEmulator(getFunctions(), 'localhost', 5001);
```

Comment:
**suggestion**: you will most likely get console errors with HMR and this trying to connect multiple times. If not then I can remove a lot of extra code from PLSS.

Reply:
I haven't had any issues. Maybe google does a better job handling multiple connects in a later version?